app/screens/thread/thread.js

```diff
@@ -25,6 +30,21 @@ export default class Thread extends Component {
 
     state = {};
 
+    componentWillMount() {
+        const {channelType, displayName, intl} = this.props;
+        let title;
+
+        if (channelType === General.DM_CHANNEL) {
+            title = intl.formatMessage({id: 'mobile.routes.thread_dm', defaultMessage: 'Direct Message Thread'});
+        } else {
+            title = intl.formatMessage({id: 'mobile.routes.thread', defaultMessage: '{channelName} Thread'}, {channelName: displayName});
+        }
+
+        this.props.navigator.setTitle({
```

Comment:
Wow, I wish I knew about this before. It always seemed really weird that whatever code switched to a new screen needed to know stuff like its title and theming. We could definitely use this elsewhere